### PR TITLE
feat: ERC20 event compliance

### PR DESCRIPTION
**Pull request by deluca-mike**
Issue: https://github.com/sherlock-audit/2023-10-mzero-judging/issues/10

### CODE REVIEW

File: src/abstract/EpochBasedInflationaryVoteToken.sol [hunk 1]
```diff
@@ -131,8 +133,12 @@ abstract contract EpochBasedInflationaryVoteToken is IEpochBasedInflationaryVote
      * @param account_ The address of the account to sync.
      */
     function _sync(address account_) internal virtual {
+        uint240 inflation_ = _getUnrealizedInflation(account_, _clock());
+
+        emit Transfer(address(this), account_, inflation_);
```
Comment (toninorair) [inline src/abstract/EpochBasedInflationaryVoteToken.sol:138]: any benefit to returning early if no unrealized inflation and do not emit useless Transfer event too?

Comment (PierrickGT) [reply src/abstract/EpochBasedInflationaryVoteToken.sol:138]: EIP20 specifies that Transfer events of 0 value must be emitted.
https://eips.ethereum.org/EIPS/eip-20#transfer-1

**Review by PierrickGT — APPROVED** [review]
Looks good. Would be good to add more tests testing the Transfer events in various conditions.

**Review by toninorair — APPROVED** [review]

File: src/abstract/EpochBasedInflationaryVoteToken.sol (hunk 1 above)
Comment (deluca-mike) [reply src/abstract/EpochBasedInflationaryVoteToken.sol:138]: It actually only says "when tokens are transfered", which implies the `transfer` and `transferFrom` functions, not a "sync". We can skip these event emissions and we'd still emit the regular `Transfer` event anyway.

**Review by PierrickGT — APPROVED** [review]

**Pull request merged**